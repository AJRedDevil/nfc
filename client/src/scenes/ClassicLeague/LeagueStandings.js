// npm packages
import React from 'react';

// our packages
import Animation from '../../components/animations';
import Table from '../../components/Table';
import LeagueStandingsPropTypes from './PropTypes';

export const getBody = standings =>
  standings.map(item => [
    item.rank,
    item.entry_name,
    item.player_name,
    item.event_total,
    item.total,
  ]);

const LeagueStandings = ({head, standings}) => (
  <Animation.Fade in>
    <div>
      <Table body={getBody(standings)} head={head} />
    </div>
  </Animation.Fade>
);
LeagueStandings.propTypes = {
  head: LeagueStandingsPropTypes.head,
  standings: LeagueStandingsPropTypes.body,
};

export default LeagueStandings;
