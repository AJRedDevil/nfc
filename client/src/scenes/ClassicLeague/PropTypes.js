import PropTypes from 'prop-types';

const head = PropTypes.arrayOf(PropTypes.shape({text: PropTypes.string}));

const body = PropTypes.arrayOf(PropTypes.object);

const classicLeagueSchema = PropTypes.shape({head});

const data = PropTypes.shape({
  leagueName: PropTypes.string.isRequired,
  creationDate: PropTypes.string.isRequired,
  lastFetched: PropTypes.string.isRequired,
  standings: PropTypes.arrayOf(PropTypes.object),
});

export default {classicLeagueSchema, data, head, body};
