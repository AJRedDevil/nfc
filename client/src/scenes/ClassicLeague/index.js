// npm packages
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {arrayOf, func, object, shape, string} from 'prop-types';
import {isEmpty} from 'lodash';

// our packages
import {fetchClassicData} from '../../services/classic/actions';
import LoadingTable from '../../components/LoadingTable';
import DateHelper from '../../utils';

class ClassicLeague extends Component {
  componentDidMount() {
    const {lastFetched, standings} = this.props.data;
    if (isEmpty(standings) || DateHelper.isAnHourAgo(lastFetched)) {
      this.props.fetchClassicData();
    }
  }

  renderTitleSubtitle = () => (
    <div>
      <h1 className="title">{this.props.data.leagueName}</h1>
      <h2 className="subtitle">
        Created : {DateHelper.ISOToDateString(this.props.data.creationDate)}
      </h2>
    </div>
  );

  renderClassicStandings = () => this.renderTitleSubtitle();

  render() {
    return isEmpty(this.props.data.standings) ? (
      <LoadingTable title="ClassicStandings" />
    ) : (
      this.renderClassicStandings()
    );
  }
}
ClassicLeague.propTypes = {
  data: shape({
    leagueName: string,
    creationDate: string,
    lastFetched: string,
    standings: arrayOf(object),
  }),
  fetchClassicData: func.isRequired,
};

const mapStateToProps = state => ({
  data: state.ClassicStandings.data,
});

const mapDispatchToProps = dispatch => ({
  fetchClassicData: () => dispatch(fetchClassicData()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ClassicLeague);
