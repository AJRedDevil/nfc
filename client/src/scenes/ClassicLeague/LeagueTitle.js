// npm packages
import React from 'react';
import PropTypes from 'prop-types';

// our packages
import {DateHelper} from '../../utils';

const LeagueTitle = ({leagueName, creationDate}) => (
  <div>
    <h1 className="title">{leagueName}</h1>
    <h2 className="subtitle">
      Created : {DateHelper.ISOToDateString(creationDate)}
    </h2>
  </div>
);
LeagueTitle.propTypes = {
  leagueName: PropTypes.string.isRequired,
  creationDate: PropTypes.string.isRequired,
};

export default LeagueTitle;
