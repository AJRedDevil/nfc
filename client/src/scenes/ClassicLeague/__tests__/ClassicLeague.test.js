// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {MemoryRouter} from 'react-router-dom';

// our packages
import ClassicLeague from '../index';

// create mock store
const mockStore = configureMockStore();

test('# ClassicLeague', () => {
  const standings = [
    {
      rank: 1,
      entry_name: 'test',
      player_name: 'test_1',
      event_total: 50,
      total: 100,
    },
    {
      rank: 2,
      entry_name: 'test',
      player_name: 'test_2',
      event_total: 80,
      total: 200,
    },
  ];
  const store = mockStore({
    ClassicStandings: {
      data: {
        leagueName: 'TestLeague',
        creationDate: '2018-02-21T00:41:36.214Z',
        lastFetched: '2018-02-21',
        standings,
      },
    },
    classicLeagueSchema: {
      head: [
        {text: 'Rank'},
        {text: 'Team'},
        {text: 'Player'},
        {text: 'GW'},
        {text: 'Total'},
      ],
    },
  });

  const component = renderer
    .create(
      <MemoryRouter>
        <ClassicLeague store={store} />
      </MemoryRouter>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
