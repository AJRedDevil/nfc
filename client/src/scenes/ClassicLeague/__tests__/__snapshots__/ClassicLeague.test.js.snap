// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# ClassicLeague 1`] = `
<div>
  <div>
    <h1
      className="title"
    >
      TestLeague
    </h1>
    <h2
      className="subtitle"
    >
      Created : 
      Wed Feb 21 2018
    </h2>
  </div>
  <div>
    <table
      className="table is-fullwidth is-striped is-narrow is-hoverable"
    >
      <thead>
        <tr />
      </thead>
      <tfoot>
        <tr />
      </tfoot>
      <tbody>
        <tr>
          <th>
            1
          </th>
          <td>
            test
          </td>
          <td>
            test_1
          </td>
          <td>
            50
          </td>
          <td>
            100
          </td>
        </tr>
        <tr>
          <th>
            2
          </th>
          <td>
            test
          </td>
          <td>
            test_2
          </td>
          <td>
            80
          </td>
          <td>
            200
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
